Processing/Explore_New_Parks.Rmd

```diff
@@ -0,0 +1,303 @@
+
+```{r}
+
+table_path <- "data/Water_Supply_Systems/NPS_Water_Systems_Database.xlsx"
+
+
+
+# Supply table (Currently filtered to BRCA)
+supply_table <- read_excel(table_path, sheet = 2, na = "NA") %>%
+  janitor::clean_names() %>%
+  dplyr::filter(region %in% c("Pacific West Region", "Intermountain Region")) %>%
+  dplyr::filter(park_unit %in% parks)
+
+# Source table
+source_table <- read_excel(table_path, sheet = 3, na = "NA") %>%
+  janitor::clean_names() %>%
+  #dplyr::filter(state == "UT")
+  dplyr::filter(wsd_system_id %in% supply_table$wsd_system_id) %>%
+  dplyr::mutate(well_depth = as.numeric(well_depth))
+
+# Copy of source table as sf_object - some columns without location data are
+# dropped. Circle back to this later.
+source_table_locs <- source_table %>%
+  drop_na(c("source_longitude", "source_longitude")) %>%
+  st_as_sf(coords = (c("source_longitude","source_latitude")), 
+           crs = 4326,
+           remove = FALSE) 
+
+park_boundaries <- getParkSystem() %>%
+  dplyr::filter(UNIT_CODE %in% parks)
+```
+
+
+Plot all park boundaries and water supply points
+```{r}
+
+a <- mapview(park_boundaries,
+          col.regions = "seagreen",
+          alpha.regions = 0.3,
+          layer.name = "Park Boundaries") +
+  mapview(source_table_locs,
+        zcol = "source_type",
+        cex = 4,
+        layer.name = "Source Type") 
+
+  leafem::addStaticLabels(a,
+                        label = park_boundaries$UNIT_CODE,
+                  noHide = TRUE,
+                  direction = 'right',
+                  textOnly = TRUE,
+                  textsize = "15px",
+                  style = list("color" = "black", "font-weight" = "bold"))
+  
+
+```
+
+Get NWIS stream gauge and well data that falls within a 0.1 degree buffer of 
+parks. We specifically want to make sure data sites have some sort of time
+series that occur after 1980, so we'll filter to only include sites with > 10
+values and for which the end date is after 1980.
+
+```{r, eval = TRUE, message = FALSE, warning = FALSE}
+
+aoi <- park_boundaries %>% st_buffer(.1)
+
+mapview(aoi)
+
+unioned_aoi <- aoi #%>%
+  #st_union() %>%
+  #st_cast(., "POLYGON") %>%
+  #st_as_sf()
+
+mapview(unioned_aoi)
+ 
+gage_sites <- vector("list", length = nrow(unioned_aoi))
+  
+  for (i in 1:nrow(unioned_aoi)){
+    
+    bbox <- sf::st_bbox(unioned_aoi[i,]) %>%
+      as.vector() %>%
+      round(., digits=7) %>% 
+      paste(collapse = ",")
+    
+    gage_sites[[i]] <- dataRetrieval::whatNWISdata(bBox = bbox)
+    
+  }
+  
+  gage_sites <- dplyr::bind_rows(gage_sites) %>%
+    dplyr::filter(year(end_date) > 1980,
+                  count_nu > 10,
+                  data_type_cd %in% c("dv", "gw"),
+                  site_tp_cd %in% c("ST","GW")) %>%
+    sf::st_as_sf(coords=c('dec_long_va','dec_lat_va'),crs=4269) %>%
+    distinct(site_no, .keep_all = TRUE) %>%
+    st_transform(st_crs(unioned_aoi)) %>%
+    st_join(park_boundaries %>% st_buffer(.2))
```

Comment:
this method is keeping all NWIS sites within the bounding box of our parks + buffer. That is, NOT the actual radius of the park boundary + buffer. Not sure if that's the intention here, but thought I'd let you know (this is one key reason the numbers aren't the same between this and the listNWIS output)

Reply:
Updated to use new listNWIS so that this isn't an issue :D